src/main/java/io/tiledb/spark/Range.java

```diff
@@ -18,7 +18,13 @@
 
   Range(Pair range) {
     this.range = range;
-    this.dataClassType = range.getFirst().getClass();
+    if (range.getFirst() != null) this.dataClassType = range.getFirst().getClass();
+    else if (range.getSecond() != null) this.dataClassType = range.getSecond().getClass();
+  }
+
+  Range(boolean empty, Pair range) {
+    this.range = range;
+    this.dataClassType = null;
```

Comment:
Is this valid? I don't think I see any checks of `dataClassType` where it uses `null`? What is supposed to happen in that case?

Reply:
Removed it, it was unnecessary.